03_letters/brvishnu92/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := `a:2
+b:1
+`
+	actual := buf.String()
+	assert.Equalf(t, expected, actual, "got %v want %v", actual, expected)
+}
+
+func TestLetters(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"normal":  {input: "smarty", expected: map[rune]int{'a': 1, 'm': 1, 'r': 1, 's': 1, 't': 1, 'y': 1}},
+		"empty":   {input: "", expected: map[rune]int{}},
+		"special": {input: "~1*{}", expected: map[rune]int{'~': 1, '1': 1, '*': 1, '{': 1, '}': 1}},
+		"emoji":   {input: "😀😎", expected: map[rune]int{'😀': 1, '😎': 1}},
+	}
+	for name, test := range testCases {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			result := letters(test.input)
+			assert.Equalf(t, test.expected, result, "got %v want %v", result, test.expected)
+		})
+	}
+}
+
+func TestSortLetters(t *testing.T) {
+	testCases := map[string]struct {
+		expected []string
+		input    map[rune]int
+	}{
+		"normal":  {expected: []string{"h:1", "k:2", "l:2", "z:2"}, input: map[rune]int{'z': 2, 'k': 2, 'h': 1, 'l': 2}},
+		"empty":   {expected: []string{}, input: map[rune]int{}},
+		"special": {expected: []string{"^:4"}, input: map[rune]int{'^': 4}},
```

Comment:
I am a little confused at what this is testing for. From the sound of the description it appears as though you are looking at the sorting of special runes, however there is really only one rune in the map so it will only ever be sorted one way, doesn't matter if it is a special or normal char. Maybe add a couple more special runes and test the sorting of such, or change this to be named "single" if that was what you were hoping to test.

Reply:
Done. Lol I was trying to show special characters was being processed but forgot about the function sorting it :-)